lighthouse-core/audits/treemap-data.js

```diff
@@ -0,0 +1,329 @@
+          name,
+          resourceBytes: unusedJavascriptSummary.totalBytes,
+          unusedBytes: unusedJavascriptSummary.wastedBytes,
+          executionTime: 0,
+        };
+      } else {
+        // TODO ...?
+        node = {
+          name,
+          resourceBytes: length,
+          unusedBytes: 0,
+          executionTime: 0,
+        };
+      }
+
+      rootNodes.push({
+        name: name,
+        node,
+      });
+    }
+
+    return rootNodes;
+  }
+
+  /**
+   * @param {LH.Artifacts.Bundle[]} bundles
+   * @param {string} url
+   * @param {LH.Artifacts['JsUsage']} JsUsage
+   * @param {LH.Audit.Context} context
+   */
+  static async getUnusedJavascriptSummary(bundles, url, JsUsage, context) {
+    const bundle = bundles.find(bundle => url === bundle.script.src);
+    const scriptCoverages = JsUsage[url];
+    if (!scriptCoverages) return;
+
+    const unusedJsSumary =
+      await UnusedJavaScriptSummary.request({url, scriptCoverages, bundle}, context);
+    return unusedJsSumary;
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<RootNode>}
+   */
+  static async makeResourceSummaryRootNode(artifacts, context) {
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const origin = new URL(artifacts.URL.finalUrl).origin;
+
+    const totalCount = networkRecords.length;
+    let totalSize = 0;
+
+    /** @type {Node[]} */
+    const children = [];
+    for (const networkRecord of networkRecords) {
+      const resourceType = ResourceSummary.determineResourceType(networkRecord);
+
+      let child = children.find(child => child.name === resourceType);
+      if (!child) {
+        child = {
+          name: resourceType,
+          resourceBytes: 0,
+          children: [],
+        };
+        children.push(child);
+      }
+
+      totalSize += networkRecord.resourceSize;
+      child.resourceBytes += networkRecord.resourceSize;
+
+      let name = networkRecord.url;
+      // TODO ...
+      if (name.startsWith(origin)) name = name.replace(origin, '/');
+      child.children = child.children || [];
+      child.children.push({
+        name,
+        resourceBytes: networkRecord.resourceSize,
+      });
+    }
+
+    return {
+      name: 'Resource Summary',
+      node: {
+        name: `${totalCount} requests`,
+        resourceBytes: totalSize,
+        children,
+      },
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    /** @type {TreemapData} */
+    const treemapData = {
+      scripts: await TreemapDataAudit.makeJavaScriptRootNodes(artifacts, context),
+      resources: [await TreemapDataAudit.makeResourceSummaryRootNode(artifacts, context)],
+    };
+
+    /** @type {LH.Audit.Details.DebugData} */
```

Comment:
Should I create a new detail type?

Reply:
> Should I create a new detail type?

it might be easiest to stay as `DebugData` during life in `experimental-config`, but when shipping to `default-config` it probably would be best as a new audit-details type since it has a real intent to use, not just debugging/httparchive debugging